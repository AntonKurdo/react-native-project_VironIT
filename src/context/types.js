export const CHANGE_THEME = 'CHANGE_THEME';
export const SET_LOCATION = 'SET_LOCATION';
export const SET_MAP_MODAL = 'SET_MAP_MODAL';
export const SHOW_LOADER = 'SHOW_LOADER';
export const HIDE_LOADER = 'HIDE_LOADER';