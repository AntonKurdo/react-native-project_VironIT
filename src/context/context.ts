import { createContext, useContext } from 'react';
import { iNewShop } from '../screens/CreateShopForm.screen';
import { LatLng } from 'react-native-maps';

type iContext = {
  state?: iState,
  coords?: LatLng,     
  changeTheme?: () => void,
  setLocation?:  (location: string, coords: LatLng ) => void,
  setMapModal?: () => void,
  showLoader?: () => void,
  hideLoader?: () => void,
  setActiveUser?: (useName: string) => void, 
  clearActiveUser?: () => void,
  setShops?:  (shop: iNewShop) => void,
  clearShops?:  () => void,
  setRadius?:  (radius: number) => void      
};

interface iState {
  activeUser: string,
  isLightenMode: boolean,
  location: null | string,
  coords: LatLng,
  isMapVisible: boolean,
  isLoader: boolean,
  shops: Array<iNewShop>,       
  radius: number
};


export const useAppContext = () => useContext(AppContext); 
const AppContext =  createContext<iContext>({});

export default AppContext;