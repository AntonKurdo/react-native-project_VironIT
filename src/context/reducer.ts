import { CHANGE_THEME, SET_LOCATION, SET_MAP_MODAL, SHOW_LOADER, HIDE_LOADER, SET_ACTIVE_USER, CLEAR_ACTIVE_USER, SET_SHOPS, CLEAR_SHOPS } from './types';

interface iCoords {
  latitude: number | null, 
  longitude: number | null
}

interface iAction {
  type: string, 
  location?: string,
  coords?: iCoords,
  userName?: string,
  shops?: object[]
};

interface iState {  
  isLightenMode: boolean,
  location: null | string,
  coords: iCoords
};

const handlers = {
  [CHANGE_THEME]: (state: iState) => ({...state, isLightenMode: !state.isLightenMode}),
  [SET_LOCATION]: (state: iState, {location, coords}: iAction) => ({...state, location, coords}),
  [SET_MAP_MODAL]: state => ({...state, isMapVisible: !state.isMapVisible}),
  [SHOW_LOADER]: state => ({...state, isLoader: true}),
  [HIDE_LOADER]: state => ({...state, isLoader: false}),
  [SET_ACTIVE_USER]: (state, {userName}) => ({...state, activeUser: userName}),
  [CLEAR_ACTIVE_USER]: (state) => ({...state, activeUser: ''}),
  [SET_SHOPS]: (state, {shops}) => ({...state, shops}),
  [CLEAR_SHOPS]: state => ({...state, shops: []})
};


export const reducer = (state: iState, action: iAction): any => { 
  return handlers[action.type](state, action);
};