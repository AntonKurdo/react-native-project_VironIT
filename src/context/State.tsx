import React, {useReducer} from 'react';
import {Context} from './context';
import {reducer} from './reducer';
import { CHANGE_THEME, SET_LOCATION, SET_MAP_MODAL, SHOW_LOADER, HIDE_LOADER, SET_ACTIVE_USER, CLEAR_ACTIVE_USER, SET_SHOPS, CLEAR_SHOPS, SET_RADIUS } from './types';

export const AppState = ({children}) => {

    const INITIAL_STATE = {   
        activeUser: '',
        isLightenMode: true,
        location: null,
        coords: {
            latitude: null, 
            longitude: null
        },
        isMapVisible: false,
        isLoader: false,
        shops: [],       
        radius: '0'
    };

    const [state, dispatch] = useReducer(reducer, INITIAL_STATE);
 
    const changeTheme : any = () => dispatch({type: CHANGE_THEME})
    const setLocation : any = (location, coords) => dispatch({type: SET_LOCATION, location, coords});
    const setMapModal : any = () => dispatch({type: SET_MAP_MODAL});
    const showLoader : any = () => dispatch({type: SHOW_LOADER});
    const hideLoader : any = () => dispatch({type: HIDE_LOADER});
    const setActiveUser : any = (userName) => dispatch({type: SET_ACTIVE_USER, userName});
    const clearActiveUser : any = () => dispatch({type: CLEAR_ACTIVE_USER});
    const setShops : any = (shops) => dispatch({type: SET_SHOPS, shops});
    const clearShops : any = () => dispatch({type: CLEAR_SHOPS}); 
    const setRadius : any = (radius) => dispatch({type: SET_RADIUS, radius}); 
 

    return <Context.Provider
        value={{
        state,
        coords: state.coords,     
        changeTheme,
        setLocation,
        setMapModal,
        showLoader,
        hideLoader,
        setActiveUser, 
        clearActiveUser,
        setShops,
        clearShops,
        setRadius       
    }}>{children}</Context.Provider>
};