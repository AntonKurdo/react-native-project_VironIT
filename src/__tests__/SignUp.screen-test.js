import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import * as AppContext from '../context/context';
import SignUpScreen from './../screens/SignUp.screen';

configure({adapter: new Adapter()});

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

const mockedNavigate = jest.fn();
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate,
    })
  };
});

describe('<SignUpScreen />' , () => {
  test('SignUpScreen render correctly', () => {    
  const contextValue = {state: {isLightenMode: true}};
  jest
    .spyOn(AppContext, 'useAppContext')
    .mockImplementation(() => contextValue);
  const wrapper = shallow(<SignUpScreen />);
  expect(wrapper).toMatchSnapshot();
  });
});