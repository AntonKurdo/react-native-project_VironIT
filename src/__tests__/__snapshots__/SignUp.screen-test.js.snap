// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpScreen /> SignUpScreen render correctly 1`] = `ShallowWrapper {}`;
